realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -208,23 +209,53 @@ std::map<std::string, int> get_enum_method(rs2::options sensor, rs2_option optio
     if (is_enum_option(sensor, option))
     {
         rs2::option_range op_range = sensor.get_option_range(option);
-        for (auto val = op_range.min; val <= op_range.max; val += op_range.step)
+        const auto op_range_min = int(op_range.min);
```

Comment:
Note that `rs2::option_range` is a `struct` with `float` values, so we should cast the `min`, `step`, `max` fields to `int` before looping. Looping with `float` values could give unexpected results due to numerical precision, especially if we accumulate the `step` in the iterator counter.

@doronhi Maybe this is the reason you were checking for the option value for enums in https://github.com/intel-ros/realsense/commit/ccaa46462a02e63033276586b20a71a9188f3fa0#diff-15761e8f7813869ae417a96398deda78R249 :thinking: